### PR TITLE
Add tokenizer_kwargs to PretrainedTransformerTokenizer.

### PR DESCRIPTION
**Pull request by hawkeoni**
Resolving issues brought up in https://github.com/allenai/allennlp/pull/3364. 
Also I hope the test run in alphabetical order just as they run on my machine, because otherwise the tests could be broken because of a bug in the transformers library https://github.com/huggingface/transformers/issues/2678.

### CODE REVIEW

**Comment by ZhaofengWu** [comment]
Looks like there's fix there already but we're only depending on released versions :(

File: allennlp/data/tokenizers/pretrained_transformer_tokenizer.py [hunk 1]
```diff
@@ -63,16 +64,19 @@ def __init__(
...
+        self.tokenizer = AutoTokenizer.from_pretrained(model_name, **tokenizer_kwargs)
 
         # Huggingface tokenizers have different ways of remembering whether they lowercase or not. Detecting it
         # this way seems like the least brittle way to do it.
         tokenized = self.tokenizer.tokenize(
             "FOO"
         )  # Use a short word that's unlikely to be cut into word pieces.
         detokenized = " ".join(tokenized)
-        self._tokenizer_lowercases = "foo" in detokenized
+        self._tokenizer_lowercases = (
```
Comment (ZhaofengWu) [inline allennlp/data/tokenizers/pretrained_transformer_tokenizer.py]: I don't believe this change is giving us anything extra, is it?

File: allennlp/data/tokenizers/pretrained_transformer_tokenizer.py [hunk 2]
```diff
@@ -63,16 +64,19 @@ def __init__(
         stride: int = 0,
         truncation_strategy: str = "longest_first",
         calculate_character_offsets: bool = False,
+        tokenizer_kwargs: Dict[str, Any] = {},
```
[the remark this answers is not in the document]
Comment (hawkeoni) [reply allennlp/data/tokenizers/pretrained_transformer_tokenizer.py]: I'm afraid that's a very implicit way to write do it. Doing it through **tokenizer_kwargs means that all the variables that do not match default variable names will be passed to tokenizer, which may cause various errors due to human inaccuracy.

File: allennlp/data/tokenizers/pretrained_transformer_tokenizer.py (hunk 1 above)
Comment (hawkeoni) [reply allennlp/data/tokenizers/pretrained_transformer_tokenizer.py]: I believe it actually changes heuristic approach, which can fail to a better one.

File: allennlp/data/tokenizers/pretrained_transformer_tokenizer.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ZhaofengWu) [reply allennlp/data/tokenizers/pretrained_transformer_tokenizer.py]: I think it's fine given the ubiquity of `**kwargs` in our code base, but I'll let @matt-gardner chime in here.

**Comment by hawkeoni** [comment]
@ZhaofengWu I'm afraid I can not properly finish this at the time, perhaps you could help?

File: allennlp/data/tokenizers/pretrained_transformer_tokenizer.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (matt-gardner) [reply allennlp/data/tokenizers/pretrained_transformer_tokenizer.py]: One thing to keep in mind: we'll want to be able to construct this using our `FromParams` pipeline.  Taking **kwargs typically for us means that you'll be passing those on to the super class.  It's probably a *lot* easier to get the `FromParams` stuff to work right if you taken it as it's done here.

**Comment by ZhaofengWu** [comment]
Okay I can take over this. Closing this and opening #3703.

**Pull request closed without merging**